### PR TITLE
Updates to tutorial and getting started

### LINKED ISSUE #12076: docs - how does prefect manage the dependencies?
### First check

- [X] I added a descriptive title to this issue.
- [X] I used GitHub search to find a similar request and didn't find it 😇

### Describe the issue

when reading the quickstart, you use the "httpx" pypi dependency.

yet there is _no_ mention at all to any dependency management on the whole page. since this is even part of the quickstart, that would be _very_ much worth mentioning. (a link might suffice)

### Describe the proposed change

add information (one line / one link might be enough) about python dependency management (note: _not_ prefect task dependency management) on deployments into prefect cloud on this page https://docs.prefect.io/latest/getting-started/quickstart/

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by discdiver**
Closes #12076

Makes minor updates for consistency in the tutorial and quickstart.
Adds Modal to list of push work pool types.
Clarifies where `httpx` comes from and what it is.

### Checklist

- [X] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [X] This pull request includes tests or only affects documentation.
- [X] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed.

For new functions or classes in the Python SDK:

- [ ] This pull request includes helpful docstrings.
- [ ] If a new Python file was added, this pull request contains a stub page in the Python SDK docs and an entry in `mkdocs.yml` navigation.

### CODE REVIEW

File: docs/getting-started/quickstart.md [hunk 1]
```diff
@@ -44,6 +44,7 @@ How can we make this script schedulable, observable, resilient, and capable of r
 pip install -U prefect
 ```
 
+Note that [httpx](https://www.python-httpx.org/) is an HTTP client library and a dependency of Prefect.
```
Comment (zzstoatzz) [inline docs/getting-started/quickstart.md:47]: this feels out of place to me

image

if we're in the middle of a high level description of what you need to get started with prefect, i don't think that taking a detour to explain one of our dependencies is necessary

maybe we could have a comment in the code on the line where we import it?

Comment (discdiver) [reply docs/getting-started/quickstart.md:47]: That's a good thought. Will update.

File: docs/getting-started/quickstart.md [hunk 2]
```diff
@@ -74,7 +74,7 @@ The fastest way to get started with Prefect is to add a `@flow` decorator to you
 [Flows](/concepts/flows/) are the core observable, deployable units in Prefect and are the primary entrypoint to orchestrated work.
 
 ```python hl_lines="2 5" title="my_gh_workflow.py"
```
Comment (chrisguidry) [inline docs/getting-started/quickstart.md:76]: Ooo nice, is this an extension that mkdocs adds, or is this Github flavored?

**Review by chrisguidry — APPROVED** [review]

**Review by zzstoatzz — APPROVED** [review]

File: docs/getting-started/quickstart.md (hunk 2 above)
Comment (discdiver) [reply docs/getting-started/quickstart.md:76]: highlighting and title are from mkdocs material.

**Pull request merged**